src/Controls/samples/Controls.Sample/Pages/Core/BrushesPage.xaml.cs

```diff
@@ -1,10 +1,97 @@
-﻿namespace Maui.Controls.Sample.Pages
+﻿using System;
+using System.Diagnostics;
+using Microsoft.Maui.Controls;
+using Microsoft.Maui.Graphics;
+
+namespace Maui.Controls.Sample.Pages
 {
 	public partial class BrushesPage
 	{
+		readonly Random _random;
+
 		public BrushesPage()
 		{
 			InitializeComponent();
+
+			BindingContext = this;
+
+			_random = new Random();
+		}
+
+		public Color Start { get; set; } = Colors.Red;
+		public Color Stop { get; set; } = Colors.Orange;
+
+		protected override void OnAppearing()
+		{
+			base.OnAppearing();
+
+			Debug.WriteLine($"BindingContextBorder Parent {BindingContextBorder.Parent}, Brush Parent {BindingContextBorder.Background.Parent}");
+		}
+
+		void OnUpdateSolidColorClicked(object sender, EventArgs e)
+		{
+			var color = GetRandomColor();
+
+			if (SolidBrushBorder.Background is SolidColorBrush solidColorBrush)
+				solidColorBrush.Color = color;
+
+			if (SolidBrushPolygon.Fill is SolidColorBrush solidBrushPolygon)
+				solidBrushPolygon.Color = color;
+		}
+
+		void OnRemovePolygonSolidColorClicked(object sender, EventArgs e)
+		{
+			BrushChangesLayout.Children.Remove(SolidBrushPolygon);
+		}
+
+		void OnUpdateLinearColorsClicked(object sender, EventArgs e)
+		{
+			var gradientStop = GetRandomGradientStop();
+			var color = GetRandomColor();
+
+			if (LinearBrushBorder.Background is LinearGradientBrush linearBrushBorder)
+				UpdateGradientStopColor(linearBrushBorder, gradientStop, color);
+
+			if (LinearBrushPolygon.Fill is LinearGradientBrush linearBrushPolygon)
+				UpdateGradientStopColor(linearBrushPolygon, gradientStop, color);
```

Comment:
Hear me out:
```suggestion
			UpdateGradientStopColor(LinearBrushBorder.Background as LinearGradientBrush, gradientStop, color);
			UpdateGradientStopColor(LinearBrushPolygon.Fill as LinearGradientBrush, gradientStop, color);
```
Later in the UpdateGradientStopColor deal with the case of it being null

Reply:
I like this suggestion, it makes it easier to read.